client/src/components/organisms/profile/profile-page.tsx

```diff
@@ -209,7 +211,10 @@ export const ProfilePage: FC<ProfilePageProps> = ({
                   title={profile.displayName}
                   rightContent={
                     !isOwnProfile && (
-                      <Button className="bg-primary hover:bg-primary/90 rounded-full text-white">
+                      <Button 
+                        className="bg-primary hover:bg-primary/90 rounded-full text-white"
+                        onClick={() => handleOnChat && handleOnChat()}

```

Comment:
add this default to the input props `handleOnChat = () => {},`

then I'm pretty sure you can just
```suggestion
                        onClick={handleOnChat}
```